### PR TITLE
Remove unused transform connections

### PR DESCRIPTION
**Pull request by tsalo**
Closes none. I'm attempting to remove any unused connections that might interfere with #681.

## Changes proposed in this pull request

- Remove template-related connections from DWI processing workflows.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -160,12 +160,14 @@ ignore = [
...
   "A001",  # TODO: Fix these
+  "A004",  # TODO: Fix these
   "B006",  # TODO: Fix these
   "S607",  # TODO: Fix these
   "S108",  # TODO: Fix these
   "S602",  # TODO: Fix these
   "E402",  # TODO: Fix these
   "UP028",  # TODO: Fix these
+  "UP031",  # TODO: Fix these
   "BLE001",  # TODO: Fix these
```
Comment (tsalo) [inline pyproject.toml:171]: I don't know why new style issues are suddenly cropping up. It looks like ruff was just missing them before. My local version isn't catching them either.

File: qsiprep/workflows/fieldmap/base.py [hunk 2]
```diff
@@ -264,16 +260,17 @@ def init_sdc_wf(fieldmap_info, dwi_meta):
...
             bold_pe=dwi_meta.get('PhaseEncodingDirection', None),
         )
 
         workflow.connect([
             (inputnode, syn_sdc_wf, [
                 ('t1_brain', 'inputnode.t1_brain'),
+                # Only place a template transform is used on DWI data
                 ('t1_2_mni_reverse_transform', 'inputnode.t1_2_mni_reverse_transform'),
                 ('b0_ref', 'inputnode.bold_ref'),
-                ('template', 'inputnode.template'),
```
Comment (tsalo) [inline qsiprep/workflows/fieldmap/base.py:276]: I just want to note this. When we switch to `--output-spaces`, we should (1) raise an error if users try to use syn with infant data and (2) only pass along the 2009c template transform here.

**Comment by tsalo** [comment]
This makes me think that we should track through the workflow connections across the package and drop any that are unused. @mattcieslak WDYT?

**Comment by mattcieslak** [comment]
I think we _will_ actually need most of these once we add the synb0-disco method

**Comment by tsalo** [comment]
I thought SynB0's requirements were basically the same as syn's. Is that not the case?

**Pull request closed without merging**